src-cpp/app/modules/main/view.cpp

```diff
@@ -15,5 +16,27 @@ void View::load()
     emit viewLoaded();
 }
 
+void View::addItem(Shared::Models::SectionItem* item)
+{
+    m_sectionModel.addItem(item);
+    emit sectionsModelChanged();
+}
+
+SectionModel* View::getSectionsModel()
+{
+    return &m_sectionModel;
+}
```

Comment:
Wow... what if i do `delete getSectionsModel()`?

Reply:
and what happens if view removed, but someone has a pointer?

Reply:
if view is owner - create it like `m_sectionModel = new SectionModel(view)` without smart pointers.